bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,47 @@
+import os
+import shutil
+
+
+class DependencyInstaller:
+    """
+    Provides functionality to copy the maven dependencies from S3 to maven local.
+    """
+
+    def __init__(self, build_id, dependency_name, version, arch):
```

Comment:
instead of "build_id, version, arch" -> use build_manfiest as input, which has all these and more. Also, you can skip dependency_name in __init__. DependencyInstaller should provide a method to sync  - all maven dependencies, or - a list of dependencies.

Reply:
Do you mean that if the user passes in the list of dependencies, then get the files for those dependencies and if nothing is passed in, get the files for all dependencies?

Reply:
Yeah 2 functions essentially - 
1. for syncing a list of maven dependencies say for a specific plugin.
2. for syncing all maven dependencies from build/ folder.

Reply:
2. can be treated optional for now. 1 is must have

Reply:
Added functionality for 1.